style_guides/architecture.md

```diff
@@ -0,0 +1,23 @@
+# Architecture Style Guide
+
+## File and Folder Structure
+
+Kibana applications, both the core project and any plugins created for Kibana, should follow this basic file and folder structure:
+
+```
+plugin_root:
+
+.
+├── public/
+├── server/
+├── common/
+└── index.js
+```
+
+The **public** folder is where you place client-side code for your application. This folder will automatically be mounted at a prefix starting with `plugins` and your application's `id`. For example, if your application's `id` is *utile*, and you were trying to import a module from `public/lib/formatter.js`, you would import modules as `plugins/utile/lib/formatter`.
```

Comment:
Do we want to encourage this style of importing when I think we're trying to get rid of our webpack aliases?